arch/arm/src/common/arm_initialize.c

```diff
@@ -28,7 +28,7 @@
 
 #include <nuttx/arch.h>
 #include <nuttx/board.h>
-#include <nuttx/sched_note.h>
+#include <nuttx/syslog/note_driver.h>
```

Comment:
Move just bofore line 37?

Reply:
And it's better to send this patch for master:
Fix note driver initialization
So we can merge the general change first and reduce the review complexity.

Reply:
I got it.
But the commit "Fix note driver initialization" cannot be moved into master as it is because file "nuttx/syslog/note_driver.h" is not exist yet in master.

This patch contains the followings:
1. include note_driver.h instead of sched_note.h
2. Remove CONFIG_SCHED_INSTRUMENTATION_BUFFER definition check because we can know whether /dev/note is available by only CONFIG_DRIVER_NOTE.

We need to divide the patch into two.
The latter one and the succeeding "Fix nxstyle warnings" (796f770) can be moved into master.

Reply:
Since the change for note_driver.h is minor, it can be incorporated into this patch too.

Reply:
The major complexity come from:
1.Redesign the buffer management
2.New IOCTL for the note driver
For these changes, we need push to feature branch before it's ready. But other change is either the bug fix or compatible enhancement, we can directly merge into mainline after review.